sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -448,20 +454,32 @@ void cancelScheduledMessage(MessagingEntityType entityType, boolean isSessionEna
      */
     @MethodSource("com.azure.messaging.servicebus.IntegrationTestBase#messagingEntityWithSessions")
     @ParameterizedTest
-    void peekFromSequenceNumberMessage(MessagingEntityType entityType, boolean isSessionEnabled) {
+    void peekFromSequenceNumberMessage(MessagingEntityType entityType, boolean isSessionEnabled) throws InterruptedException {
         // Arrange
         final int entityIndex = 3;
 
         setSender(entityType, entityIndex, isSessionEnabled);
         final String messageId = UUID.randomUUID().toString();
         final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
 
-        sendMessage(message).block(TIMEOUT);
+        sendMessage(message).block();
 
         setReceiver(entityType, entityIndex, isSessionEnabled);
 
-        final ServiceBusReceivedMessage peekMessage = receiver.peekMessage().block(TIMEOUT);
-        assertNotNull(peekMessage);
+        // Message are not always guaranteed, so try many times
+        final ServiceBusReceivedMessage peekMessage = receiver.peekMessage()
+            .filter(receivedMessage -> messageId.equals(receivedMessage.getMessageId()))
+            .flatMap(receivedMessage -> {
+                countDownLatch.countDown();
+                return Mono.just(receivedMessage);
+            })
```

Comment:
Is this same as below?
```suggestion
            .map(receivedMessage -> {
                countDownLatch.countDown();
                return receivedMessage;
            })
```